connectors/src/connectors/slack/temporal/workflows.ts

```diff
@@ -126,3 +135,91 @@ export async function workspaceSyncOneChannel(
   );
   console.log(`Syncing channel ${channelName} (${channelId}) done`);
 }
+
+export async function syncOneThreadDebounced(
+  connectorId: string,
+  dataSourceConfig: DataSourceConfig,
+  nangoConnectionId: string,
+  channelId: string,
+  threadTs: string
+) {
+  let signaled = false;
+  let debounceCount = 0;
+
+  setHandler(newWebhookSignal, () => {
+    console.log("Got a new webhook ");
+    signaled = true;
+  });
+
+  while (signaled) {
+    signaled = false;
+    await sleep(10000);
+    if (signaled) {
+      debounceCount++;
+      continue;
+    }
+    const slackAccessToken = await getAccessToken(nangoConnectionId);
+    const channel = await getChannel(slackAccessToken, channelId);
+    if (!channel.name) {
+      throw new Error(`Could not find channel name for channel ${channelId}`);
+    }
+
+    console.log(`Talked to slack after debouncing ${debounceCount} time(s)`);
+    await syncThread(
+      dataSourceConfig,
+      slackAccessToken,
+      channelId,
+      channel.name,
+      threadTs,
+      connectorId
+    );
+  }
+  // /!\ Any signal received outside of the while loop will be lost, so don't make any async
```

Comment:
Your comment hint that this could happen in the middle of the two async calls above? Is that the case?

Reply:
Ok now I understand. Was thinking the async part was after the loop 👍